### ISSUE TITLE
Implement WAI-ARIA 2.0 accesibility roles.

### ISSUE THREAD

**Issue by Giwayume** [Semantic-UI-Vue/Semantic-UI-Vue#302]
It is important that a Javascript UI framework automates accessibility as much as it can and makes recommendations to its users in the documentation where manual action must be taken in order to assure accessibility compliance.

[Bootstrap Vue is a shining example of this.](https://bootstrap-vue.js.org/docs/components/form-group)

Companies with online storefronts can and have been sued for not having accessible websites under the  Americans with Disabilities Act (ADA).

The first thing I check to see if a UI framework is accessible is if the modal implementation places `role="dialog"` on the modal. Does the Escape key close the modal? Is focus trapping inside of the modal implemented, or can I just tab out of the modal on to the page behind the dimmer and keep interacting with the page behind the modal while it's open?

There's a lot of functional accessibility behavior that Semantic UI is lacking in all of its Javascript implementations. I like using Vue, so I'm bringing it up here.
